### PR TITLE
User Thresholding

### PR DESCRIPTION
**Pull request by nathanielparke**
Allows a user to set a threshold for classification in the logistic regression suite for EvaluateModel.

### CODE REVIEW

**Comment by devin-petersohn** [comment]
pinging @fnothaft @heuermh to approve Jenkins build test.

File: gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala [hunk 1]
```diff
@@ -68,5 +68,52 @@ class LogisticValidationRegressionSuite extends GnocchiFunSuite {
     }).count(b => { b }))
     assert(predictionResult.sameElements(expectedResults))
   }
+
+  sparkTest("Test logistic regression prediction on binary data with a different threshold") {
+    // Reads in the binaryClean data, which is the same as the above data but removes any
+    // duplicate samples. The Fourth row (0 indexed) is the prediction results generated
+    // from a logisitic regression model in excel with a threshold classification value of 0.3
+    val pathToFile = ClassLoader.getSystemClassLoader.getResource("binaryThreshold3.csv").getFile
```
Comment (heuermh) [inline gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala]: bdg-utils `SparkFunSuite` provides a `testFile` method

https://github.com/bigdatagenomics/utils/blob/master/utils-misc/src/test/scala/org/bdgenomics/utils/misc/SparkFunSuite.scala#L116

File: gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala [hunk 2]
```diff
@@ -68,5 +68,52 @@ class LogisticValidationRegressionSuite extends GnocchiFunSuite {
...
+    val statistics = Map("weights" -> weights)
+    val assoc = Association(fakeVariant, "", 0.0, statistics)
+
+    // Array[(String, Double)] where String is sampleid and Double is predicted value
+    val predictionResult = AdditiveLogisticEvaluation.predictSite(sampleObservations, assoc, 0.3)
+
+    // Assert that the predictions result in the same as the actual phenotype (on dummy set)
+    for (i <- predictionResult.indices) {
+      if (predictionResult(i) != expectedResults(i)) {
+        print("Error --> ")
```
Comment (heuermh) [inline gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala]: these debugging statements shouldn't be left in, make assertions and let the test framework report errors

**Review by heuermh — CHANGES_REQUESTED** [review]

**Comment by heuermh** [comment]
Jenkins, add to whitelist

**Pull request closed without merging**